src/main/java/duke/util/Parser.java

```diff
@@ -0,0 +1,122 @@
+package duke.util;
+
+import duke.command.*;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+
+/**
+ *  Represents the parser of the program that converts user inputs into Commands.
+ */
+public class Parser {
+    public enum Cmd{
+        list, todo, deadline, event, mark, unmark, delete, find, bye, none;
+    }
+    /**
+     * Initializes the given Task ArrayList with given input,
+     * by adding the tasks from each line of input into the ArrayList.
+     * The different data in each line of a stored task is splitted by a "|" sign.
+     * THe method processes these data, creates corresponding task and add the ArrayList accordingly.
+     *
+     * @param input the input for each task, the input typically comes from a stored data,
+     *              accessed by the Storage class.
+     * @param taskList the given Task ArrayList to be initialized.
+     */
+    public static void initializeTask(String input, ArrayList<Task> taskList) {
+        String[] data = input.split(" \\| ");
+        Task task = new Task("");
+        switch (data[0]) {
+            case "T":
+                task = new Todo(data[2]);
+                break;
+            case "D":
+                LocalDateTime date = LocalDateTime.parse(data[3], DateTimeFormatter.ofPattern("yyyy/MM/dd HHmm"));
+                task = new Deadline(data[2], date);
+                break;
+            case "E":
+                LocalDateTime fromDate = LocalDateTime.parse(data[3], DateTimeFormatter.ofPattern("yyyy/MM/dd HHmm"));
+                LocalDateTime toDate = LocalDateTime.parse(data[4], DateTimeFormatter.ofPattern("yyyy/MM/dd HHmm"));
+                task = new Event(data[2], fromDate, toDate);
+                break;
+            default:
+                System.out.println("PROBLEM encountered with the saved data while loading");
+                break;
+        }
```

Comment:
The indentation seems to violate coding styles?

```suggestion
        switch (data[0]) {
        case "T":
            task = new Todo(data[2]);
            break;
        case "D":
            LocalDateTime date = LocalDateTime.parse(data[3], DateTimeFormatter.ofPattern("yyyy/MM/dd HHmm"));
            task = new Deadline(data[2], date);
            break;
        case "E":
            LocalDateTime fromDate = LocalDateTime.parse(data[3], DateTimeFormatter.ofPattern("yyyy/MM/dd HHmm"));
            LocalDateTime toDate = LocalDateTime.parse(data[4], DateTimeFormatter.ofPattern("yyyy/MM/dd HHmm"));
            task = new Event(data[2], fromDate, toDate);
            break;
        default:
            System.out.println("PROBLEM encountered with the saved data while loading");
            break;
        }
```